### PR TITLE
Add u256 multiplication to standard library

### PR DESCRIPTION
**Pull request by bobbinth**
This PR adds u256 multiplication procedure to the standard library.

### CODE REVIEW

**Review by grjte — APPROVED** [review]
Looks good! I left a couple small things inline.

File: stdlib/asm/math/u256.masm [hunk 1]
```diff
@@ -197,4 +197,343 @@ export.eq_unsafe
...
+    swapw
+    storew.local.3
+    # x[5-8] at 3 #
+    padw
+    storew.local.4
+    storew.local.5
+    # p at 4 and 5 #
+
+    # y[0] #
+    dropw
```
[the remark this answers is not in the document]
Comment (grjte) [reply stdlib/asm/math/u256.masm:268]: Maybe that should be removed from the local io tests, then?

File: stdlib/asm/math/u256.masm [hunk 2]
```diff
@@ -197,4 +197,343 @@ export.eq_unsafe
...
+    exec.mulstep
+    swap
+    movdn.6
+end
+
+# Performs addition of two unsigned 256 bit integers discarding the overflow. #
+# The input values are assumed to be represented using 32 bit limbs, but this is not checked. #
+# Stack transition looks as follows: #
+# [b7, b6, b5, b4, b3, b2, b1, b0, a7, a6, a5, a4, a3, a2, a1, a0, ...] -> [c7, c6, c5, c4, c3, c2, c1, c0, ...] #
+# where c = (a + b) % 2^256, and a0, b0, and c0 are least significant 32-bit limbs of a, b, and c respectively. #
```
[the remark this answers is not in the document]
Comment (bobbinth) [reply stdlib/asm/math/u256.masm]: fixed!

File: stdlib/asm/math/u256.masm (hunk 1 above)
[the remark this answers is not in the document]
Comment (bobbinth) [reply stdlib/asm/math/u256.masm]: Yes - good point. I'll add this to the assembly specs. Would you mind updating the tests?

[the remark this answers is not in the document]
Comment (grjte) [reply stdlib/asm/math/u256.masm]: Sure, I'll update them

**Pull request merged**